document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/LogicExpression.java

```diff
@@ -39,25 +39,25 @@
 @Value
 @Builder
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
-public class LogicalExpression implements FilteringExpression {
+public class LogicExpression implements FilterTypeExpression {
```

Comment:
I think `LogicalExpression` is more easily understandable.
Look up terminology from https://www.geeksforgeeks.org/what-is-an-expression-and-what-are-the-types-of-expressions/